scenarios/state_transitions_steady.go

```diff
@@ -114,10 +119,29 @@ func (s *stateTransitionsSteady) run(ctx context.Context) error {
 			}(iter)
 		}
 	}
-	// We don't want to wait to let workflows complete here. In dev cases the
-	// local server will just shutdown, in non-dev cases, the server admin can
-	// wait. But we don't want to introduce delays for those that may want to run
-	// this quickly/frequently.
-	s.Logger.Infof("Run complete, ran %v iterations", iter)
-	return nil
+
+	// We are waiting to let workflows complete here, knowing that this part of
+	// the scenario is not steady state transitions. We will wait a minute max to
+	// confirm all workflows on the task queue are no longer running.
+	s.Logger.Infof("Run complete, ran %v iterations, waiting on all workflows to complete", iter)
+	// First, wait for all starts to have started (they are done in goroutine)
+	startWG.Wait()
+	// Now check every so often
+	start := time.Now()
+	const interval = 3 * time.Second
+	const maxWait = time.Minute
+	for {
+		resp, err := s.Client.CountWorkflow(ctx, &workflowservice.CountWorkflowExecutionsRequest{
```

Comment:
Heh, so, like I mentioned in the other PR, already a use-case for re-using that function.

Reply:
Maybe a more general "EnsureNoRunningWorkflows"